theano/gof/utils.py

```diff
@@ -217,22 +217,24 @@ def __str__(self):
     def __call__(cls, *args, **kwargs):
         # Rules of caching:
         #   - when the props is None, we never cache
-        #   - when the props is (), we always cache
-        #   - when the props is defined but not empty,
+        #   - when the props is (), the init should only contain self.
+        #   - when the props is defined and is none empty:
         #     the args and kwargs should only contain the porps
         # Note: when the props is empty inspect.getargspec(cls.__init__)
         #       gives an error since cls.__init__ is not a python function.
 
         props = cls.__dict__.get('__props__', None)
-
-        # The cases are being separated for code readability.
+        try:
+            init_args = sorted(inspect.getargspec(cls.__init__)[0][1:])
+        except:
+            init_args = None
```

Comment:
If the __init__ was defined in the super, inspect can find it. I had the exceptions when the init was a slot wrapper. Thus it was not a function and inspect could not find it.